src/Bathymetry.jl

```diff
@@ -189,26 +179,44 @@ function regrid_bathymetry(target_grid;
 
     target_z = interpolate_bathymetry_in_passes(native_z, target_grid; 
                                                 passes = interpolation_passes,
-                                                connected_regions_allowed,
+                                                major_basins,
                                                 minimum_depth)
 
+    if minimum_depth > 0
+        zi = interior(target_z, :, :, 1)
+        shallow_ocean = zi .> - minimum_depth
+        zi[shallow_ocean] .= 0
+    end
```

Comment:
can we do this on the GPU? Or do we need a kernel?

Reply:
I would have thought it works on the GPU but I will check and write a kernel if not

Reply:
it works

```
julia> a = CuArray(rand(4, 4))
4×4 CuArray{Float64, 2, CUDA.Mem.DeviceBuffer}:
 0.968109  0.435424  0.988271  0.108635
 0.102944  0.956011  0.637797  0.140668
 0.466575  0.800952  0.397086  0.812225
 0.221806  0.308943  0.385949  0.471147

julia> small = a .< 0.5
4×4 CuArray{Bool, 2, CUDA.Mem.DeviceBuffer}:
 0  1  0  1
 1  0  0  1
 1  0  1  0
 1  1  1  1

julia> a[small] .= 0
10-element view(::CuArray{Float64, 2, CUDA.Mem.DeviceBuffer}, CartesianIndex{2}[CartesianIndex(2, 1), CartesianIndex(3, 1), CartesianIndex(4, 1), CartesianIndex(1, 2), CartesianIndex(4, 2), CartesianIndex(3, 3), CartesianIndex(4, 3), CartesianIndex(1, 4), CartesianIndex(2, 4), CartesianIndex(4, 4)]) with eltype Float64:
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
```

Reply:
nice

Reply:
this also work son the GPU for in-place

```
julia> map!(x -> x * (x < 0.5), a, a)
4×4 CuArray{Float64, 2, CUDA.Mem.DeviceBuffer}:
 0.0       0.403628  0.0       0.00580952
 0.0       0.393159  0.335011  0.263924
 0.481604  0.0       0.0       0.0
 0.124625  0.0       0.452182  0.399641
```

Reply:
actually this is better 

```julia
julia> a .*= a .> b
4×4 CuArray{Float64, 2, CUDA.Mem.DeviceBuffer}:
 0.775472  0.0       0.0  0.0
 0.0       0.838647  0.0  0.0
 0.502095  0.0       0.0  0.0
 0.0       0.74245   0.0  0.639991
```